lib/lita_versioner/project_repo.rb

```diff
@@ -112,35 +123,69 @@ def has_file?(path_from_repo_root)
     # Time since the last commit on `git_ref` in seconds
     def time_since_last_commit_on(git_ref)
       now = Time.new.to_i
-      commit_time = run_command("git show -s --format=\"%ct\" #{git_ref}").stdout.strip.to_i
+      commit_time = run_command(%W{git show -s --format=\"%ct\" #{git_ref}}).stdout.strip.to_i
       now - commit_time
     end
 
-    def run_command(command, cwd: repo_directory)
-      handler.run_command(command, cwd: cwd)
+    def current_sha
+      run_command(%w{git show-ref master}).stdout.split(/\s/).first
     end
 
-    def repo_directory
-      File.join(CACHE_DIRECTORY, repo_name)
+    def run_command(command, cwd: repo_directory)
+      handler.run_command(command, cwd: cwd)
     end
 
     def repo_name
       # Note that this matches github urls both like:
       #   https://github.com/litaio/development-environment.git
       #   git@github.com:chef-cookbooks/languages.git
-      github_url.match(/.*\/(.*)\.git$/)[1]
+      #   /tmp/lita-test
+      repo = File.basename(github_url)
+      repo = repo[0..-4] if repo.end_with?(".git")
+      repo
     end
 
     def ensure_git_config_set
-      if !run_command("git config -l").stdout.match(/chef-versioner@chef.io/)
-        run_command("git config user.email \"chef-versioner@chef.io\"")
-        run_command("git config user.name \"Chef Versioner\"")
+      if !run_command(%w{git config -l}).stdout.match(/chef-versioner@chef.io/)
+        run_command(%w{git config user.email chef-versioner@chef.io})
+        run_command(%w{git config user.name} + [ "Chef Versioner" ])
       end
     end
 
     def ensure_cache_dir_exists
       Dir.mkdir(CACHE_DIRECTORY) unless File.exist? CACHE_DIRECTORY
     end
 
+    private
+
+    # Clone ./cache/chef-dk.git into ./cache/<handler.id>/chef-dk
+    def clone
+      # Grab ./cache/chef-dk.git.
+      refresh_main_repository
+
+      # Ensure ./cache/<handler.id> exists.
+      FileUtils.mkdir_p(File.dirname(repo_directory))
```

Comment:
No longer need this mkdir_p - repo_directory should call sandbox_directory which should initialize that for us.